contracts/impl/EnigmaCommon.sol

```diff
@@ -101,4 +119,20 @@ library EnigmaCommon {
     {
         return _message.concat(uint64(_arraylength).toBytesFromUint64());
     }
+
+    /**
+    * Append the length of an array to an existing bytes buffer
+    *
+    * @param _message Bytes buffer being appended to
+    * @param _arraylength Length of array
+    * @param _typeLength Length of each item in bytes
+    * @return New bytes buffer
+    */
+    function appendMessageArrayLengthKM(uint256 _arraylength, uint256 _typeLength, bytes memory _message)
+    internal
+    pure
+    returns (bytes memory)
+    {
+        return (_message.concat(hex"01")).concat(uint64(_arraylength.mul(_typeLength.add(9))).toBytesFromUint64());
```

Comment:
I think that the hardcoded solution is never good. It should be a size of byte and size of `uint64`.